### ISSUE TITLE
Raw jax.grad through a stiff network silently returns non-finite without dtmax

### ISSUE THREAD

**Issue by ctdegroot** [ctdegroot/aquakin#192]
## Surprise
A hand-rolled reverse-mode gradient through a stiff network silently returns NaN/inf when no `dtmax` cap is set:

```python
g = jax.grad(lambda p: reactor.solve(C0, p, t_span=...).C[-1].sum())(params)  # ASM/ADM/WATS -> non-finite
```

No exception is raised. `calibrate(check_finite=True)` and `sensitivity(check_finite=True)` guard this, but a user who writes their own loss + optimizer through `reactor.solve()` is fully exposed.

## Why it bites
The gradient flows into an optimizer as silent garbage; the run "works" but never converges, with no indication the gradient was non-finite.

## Discoverability
The `dtmax` remedy is in the `BatchReactor` docstring and CLAUDE.md, but nothing at the raw `jax.grad` call site flags it.

## Suggested remedy
- Document the raw-`jax.grad` exposure in the reactor docstring (point users at `dtmax`, forward mode, or `calibrate`/`sensitivity`).
- Optionally expose a lightweight `reactor.check_gradient_finite(...)` helper mirroring `check_finite_gradient`.

Files: `aquakin/integrate/_common.py` (`check_finite_gradient`), `aquakin/integrate/batch.py` (docstring).
